rbac/management/audit_log/model.py

```diff
@@ -96,15 +98,16 @@ def get_resource_item(self, r_type, request, *args, **kwargs):
             return None
 
         elif r_type == "principal":
+            print("entering principal")
             principal_object = get_object_or_404(Principal, username=request.user.username, tenant=verify_tenant)
```

Comment:
After the tests in a fresh database, @Ellen-Yi-Dong and I found the reason PR_Checks is not passing. It is related to getting the principals from the database once the user requester is not a principal yet.
The user requester just will be inside the database as a principal just after being inside a group, so when we are creating a group using a user requester that is not a principal yet, we are getting 404.

Reply:
Attempting to fix this issue from this ticket: https://issues.redhat.com/browse/RHCLOUD-33727

Will create separate PR that will most likely be merged first